include/lapack/gelq2.hpp

```diff
@@ -60,24 +61,37 @@ namespace tlapack
         using idx_t = size_type<matrix_t>;
         using range = std::pair<idx_t, idx_t>;
 
+        // constants
         const idx_t m = nrows(A);
         const idx_t n = ncols(A);
         const idx_t k = std::min(m, n);
 
+        // check arguments
+        tlapack_check_false(access_denied(dense, write_policy(A)), -1);
+        tlapack_check_false((idx_t)size(tauw) < std::min<idx_t>(m, n), -2);
+        tlapack_check_false((idx_t)size(work) < m, -3);
+
+        // quick return
+        if (m <= 0)
```

Comment:
@weslleyspereira Hey Weslley, I referred to your geqr2.hpp here and changed the quick return index `n` to `m` since the `work` for gelq2 is always `m`. Is this okay here?